prow/plugins/project/project.go

```diff
@@ -0,0 +1,348 @@
+	errListingReposMsg        = "Error listing all repos in org %s."
+	errListingRepoProjectsMsg = "Error listing all projects under repo %s."
+	errListingOrgProjectsMsg  = "Error listing all projects under org %s."
+	errRemovingFromProjectMsg = "Error removing project card from project %s (ID %d) for %s/%s#%d."
+	errListingColsMsg         = "Error listing all columns under project %s."
+	errMovingCardMsg          = "Error moving project card to a new column %s (ID %d) in the same project (%s)."
+	errCreatingCardMsg        = "Error creating project card for project %s column %s (ID %d) for %s/%s#%d."
+	successMovingCardMsg      = "You have successfully moved the project card for this issue to column %s (ID %d)."
+	successCreatingCardMsg    = "You have successfully created a project card for this issue. It's been added to project %s column %s (ID %D)."
+	successClearingProjectMsg = "You have successfully removed this issue/PR from project %s."
+	clearKeyword              = "clear"
+	projectNameToIDMap        = make(map[string]int)
+)
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	ListTeamMembers(id int, role string) ([]github.TeamMember, error)
+	GetRepos(org string, isUser bool) ([]github.Repo, error)
+	GetRepoProjects(owner, repo string) ([]github.Project, error)
+	GetOrgProjects(org string) ([]github.Project, error)
+	GetProjectColumns(projectID int) ([]github.ProjectColumn, error)
+	CreateProjectCard(columnID int, projectCard github.ProjectCard) (*github.ProjectCard, error)
+	GetColumnProjectCard(columnID int, cardNumber int) (*github.ProjectCard, error)
+	MoveProjectCard(projectCardID int, newColumnID int) error
+	DeleteProjectCard(projectCardID int) error
+	TeamHasMember(teamID int, memberLogin string) (bool, error)
+}
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func getMaintainerTeam(pluginConfigs plugins.ProjectConfig, org string, repo string) (int, string) {
+	for orgName, orgConfig := range pluginConfigs.Orgs {
+		if org == orgName {
+			// look for repo level configs first because repo level config overrides org level configs
+			for repoName, repoConfig := range orgConfig.Repos {
+				if repo == repoName {
+					return repoConfig.MaintainerTeamID, repoConfig.MaintainerTeamName
+				}
+			}
+			return orgConfig.MaintainerTeamID, orgConfig.MaintainerTeamName
+		}
+	}
+	return 0, ""
+}
+
+func getColumnMap(pluginConfigs plugins.ProjectConfig, org string, repo string) map[string]string {
+	for orgName, orgConfig := range pluginConfigs.Orgs {
+		if org == orgName {
+			for repoName, repoConfig := range orgConfig.Repos {
+				if repo == repoName {
+					return repoConfig.ProjectColumnMap
+				}
+			}
+			return orgConfig.ProjectColumnMap
+		}
+	}
+	return nil
+}
+
+func getOrgColumnMap(pluginConfigs plugins.ProjectConfig, org string, repo string) map[string]string {
```

Comment:
Nit: this could be a method on `ProjectConfig` and does not need the `repo` parameter